core/state_transition.go

```diff
@@ -576,6 +611,14 @@ func (st *StateTransition) creditTxFees() error {
 		baseTxFee = new(big.Int)
 	}
 
+	if st.msg.MaxFeeInFeeCurrency() != nil {
+		// Celo Denominated
+		refund = feeCurrencyRate.FromBase(refund)
+		tipTxFee = feeCurrencyRate.FromBase(tipTxFee)
+		baseTxFee = feeCurrencyRate.FromBase(baseTxFee)
```

Comment:
This might not add up to the debited amount due to the integer division in `FromBase`. For that reason, the current CIP-66 draft specifies
```
feeInFeeCurrency = baseFeeInFeeCurrency + tipInFeeCurrency
refundInFeeCurrency = debitInFeeCurrency - feeInFeeCurrency
```

Reply:
so I should be able to exchange the tip & base, but not the refund, correct? check the new version of the code

Reply:
Yes, you can exchange both tip and base, just not all three values. I'll comment on the new code separately.